span-stacktrace/src/main/java/io/opentelemetry/contrib/stacktrace/StackTraceSpanProcessor.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.stacktrace;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.contrib.stacktrace.util.AbstractSimpleChainingSpanProcessor;
+import io.opentelemetry.contrib.stacktrace.util.MutableSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.function.Function;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class StackTraceSpanProcessor extends AbstractSimpleChainingSpanProcessor {
+
+  // TODO : remove this once semconv 1.24.0 is available
+  static final AttributeKey<String> SPAN_STACKTRACE = AttributeKey.stringKey("code.stacktrace");
+
+  private static final Logger logger = Logger.getLogger(StackTraceSpanProcessor.class.getName());
+
+  private final long minSpanDurationNanos;
+
+  private final Function<ReadableSpan, Boolean> filterFunction;
```

Comment:
[Predicate](https://docs.oracle.com/javase/8/docs/api/java/util/function/Predicate.html) should be the preferred standard type here